### PR TITLE
Prepare for IntelliJ EAP 2021.1

### PR DESCRIPTION
**Pull request by Flanker32**
Origin PR : https://github.com/microsoft/azure-tools-for-java/pull/4976

Fixes:
- Migrate LinkAzureServiceRunConfigurationExtension to EAP interface
- Using IconLoader.getIcon in case call classloader with leading slash issue,  fixes https://dev.azure.com/mseng/VSJava/_workitems/edit/1824447/
- Implement getFile in FileEditors, fixes https://dev.azure.com/mseng/VSJava/_workitems/edit/1824452, https://dev.azure.com/mseng/VSJava/_workitems/edit/1824453
- Fix logic to get terminal input stream in EAP, fixes https://dev.azure.com/mseng/VSJava/_workitems/edit/1824462/

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/gradle.properties [hunk 1]
```diff
@@ -2,7 +2,7 @@ javaVersion=11
 org.gradle.jvmargs='-Duser.language=en'
 sources=false
 intellij_version=IC-LATEST-EAP-SNAPSHOT
-dep_plugins=org.intellij.scala:2020.3.8
+dep_plugins=org.intellij.scala:2021.1.10
 applicationinsights.key=57cc111a-36a8-44b3-b044-25d293b8b77c
 updateVersionRange=true
-patchPluginXmlSinceBuild=203.5419.21
+patchPluginXmlSinceBuild=211.6085.26
```
Comment (t-rufang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/gradle.properties:8]: New EAP version is released. We should do the following version upgrade.
1. Upgrade Intellij scala plugin version to `2021.1.12` based on [plugin version description](https://plugins.jetbrains.com/plugin/1347-scala/versions/eap)
2. Upgrade lower bound of IntelliJ version to `211.6432.7` based on [official site description](https://www.jetbrains.com/idea/nextversion/#section=windows)

File: gradle.properties [hunk 2]
```diff
@@ -1,9 +1,9 @@
 ARTIFACTS_DIR=artifacts
 INJECT_INTELLIJ_VERSION=true
-IJ_VERSION_MAJOR_BUILD=203.5419.21
-IJ_VERSION_MAJOR=2020.3
+IJ_VERSION_MAJOR_BUILD=211.6085.26
+IJ_VERSION_MAJOR=LATEST-EAP-SNAPSHOT
 IJ_VERSION_MINOR=
-IJ_SCALA_VERSION_LATEST=2020.3.8
+IJ_SCALA_VERSION_LATEST=2021.1.10
 skipCheckstyle=true
```
Comment (t-rufang) [inline gradle.properties:7]: Same version upgrade here

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/PluginUtil.java [hunk 3]
```diff
@@ -155,7 +155,7 @@ public static String getPluginRootDirectory() {
     }
 
     public static Icon getIcon(String iconPath) {
-        return IconLoader.getIcon(iconPath);
+        return IconLoader.getIcon(iconPath, PluginUtil.class);
```
Comment (t-rufang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/PluginUtil.java:158]: Great to see we get this issue fixed~

**Review by t-rufang — COMMENTED** [review]
Thanks Hanxiao for the fix~

**Review by andxu — APPROVED** [review]

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**